cli/tools/upgrade.rs

```diff
@@ -348,12 +349,27 @@ pub async fn upgrade(upgrade_flags: UpgradeFlags) -> Result<(), AnyError> {
   check_exe(&new_exe_path)?;
 
   if !upgrade_flags.dry_run {
-    match upgrade_flags.output {
-      Some(path) => {
-        fs::rename(&new_exe_path, &path)
-          .or_else(|_| fs::copy(&new_exe_path, &path).map(|_| ()))?;
+    let output_result = match upgrade_flags.output {
+      Some(path) => fs::rename(&new_exe_path, &path)
+        .or_else(|_| fs::copy(&new_exe_path, &path).map(|_| ())),
+      None => replace_exe(&new_exe_path, &current_exe_path),
+    };
+    if let Err(err) = output_result {
+      const WIN_ERROR_ACCESS_DENIED: i32 = 5;
+      if cfg!(windows) && err.raw_os_error() == Some(WIN_ERROR_ACCESS_DENIED) {
+        return Err(err).with_context(|| {
+          format!(
+            concat!(
+            "Could not replace the old executable. ",
+            "Please ensure there are no running deno processes (ex. Stop-Process -Name deno ; deno {}) ",
+            "and close any editors before upgrading."
```

Comment:
Maybe you should say something about permission here. Like
> Access Denied: Could not replace the old Deno executable. Please ensure there are no running deno processes (ex. Stop-Process -Name deno ; deno {}) and close any editors before upgrading.

Reply:
Good point. Updated.